rusty-machine/src/learning/gmm.rs

```diff
@@ -132,6 +132,36 @@ impl GaussianMixtureModel {
         }
     }
 
+    /// Constructs a new GMM with the specified prior mixture weights.
+    ///
+    /// The mixture weights must have the same length as the number of components.
+    /// The sum of the mixture weights must be 1.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use rusty_machine::learning::gmm::GaussianMixtureModel;
+    /// use rusty_machine::linalg::vector::Vector;
+    ///
+    /// let mix_weights = Vector::new(vec![0.25, 0.25, 0.5]);
+    ///
+    /// let _ = GaussianMixtureModel::with_weights(3, mix_weights);
+    /// ```
+    pub fn with_weights(k: usize, mixture_weights: Vector<f64>) -> GaussianMixtureModel {
+        assert!(mixture_weights.size() == k, "Mixture weights must have length k.");
```

Comment:
I'm not too sure about this.. I don't think an `Option` is appropriate here. Result may be but I tend to consider those to be for recoverable errors. I don't see a situation where a Result would be more helpful than panicking?

Of course - detecting this at compile time would be ideal. But I'm unconvinced that letting the user face the panic when they unpack the result is any better? (If anything it makes the library a tiny bit more annoying to use?)

Please feel free to argue back! I should definitely be using Results in more places...